### DISCUSSION TITLE
FSLeyes: Feedback as a user

### DISCUSSION THREAD

**Discussion by mathieuboudreau** [neuropoly/axondeepseg#450] [Show and tell]
Hi folks!

Last week I spent about 15-20 hours segmenting some images using the FSLeyes GUI, and thought I'd give back some feedback from a users point of view.

* I encountered a user scenario that has a high risk of resulting in data error. Currently, if a user loads an image in FSLeyes and then begins the segmentation, if a pixel size file doesn't exist the user will be prompted to enter the resolution. As we've suggested to some users, and I had to do myself, tweaking this parameter is sometimes necessary to adjust the patch size to improve the segmentations. **However**, if after the segmentation is finished the user clicks the save morphometrics button within the same session, **the user will not be prompted to enter the actual resolution, thus the values in the morphometrics file will be wrong (except for g-ratio)**. I caught it, but users likely wouldn't. My solution was to generate the segmentations for each images, close FSLeyes, and then load images one by one to manually fix them. I think we urgently need a solution to both of these problems mentioned above.

* This is more of an FSLeyes quirk that I encountered than ADS but thought I'd mention it, but it's related to another ADS quirk. Sometimes, due to using patches, our ADS segmentation images results in exactly horizontal lines (approx. 1 pixel wide I think) or vertical. I notice that these lines don't always show up in FSLeyes, and will only appear if you adjust the zoom to certain values. FSLeyes must be skipping some pixels when displaying to the user. The major problem with this is that if the user sets a certain zoom value and then corrects the masks manually, then the final corrected masks may include entire lines through axons, which is very annoying. See below for an example (look at the large axon in the top right).

Zoom value: 1509.2

<img width="827" alt="Capture d’écran 2021-02-02 à 16 08 53" src="https://user-images.githubusercontent.com/1421029/106653173-0389bb80-6571-11eb-97c3-9bf902945702.png">

Zoom value: 1502.7

<img width="827" alt="Capture d’écran 2021-02-02 à 16 09 05" src="https://user-images.githubusercontent.com/1421029/106653181-071d4280-6571-11eb-81c8-b6eec80eb30d.png">

* After manual correction, anytime that there is overlap between myelin and axons a popup error appears (see below) to warn the user about the behaviour. This sounded like a good idea initially, but I can tell you that after seeing this error about 30 times, it became extremely annoying. Maybe we can just show it once to the user and then stop, or don't show it at all and we can document it in the documentation.

<img width="1116" alt="Capture d’écran 2021-02-02 à 16 13 15" src="https://user-images.githubusercontent.com/1421029/106653935-03d68680-6572-11eb-8d7a-acaa6590b19f.png">

**Comment by jcohenadad** [comment]
Thank you so much for your feedback @mathieuboudreau . A software can be so much improved when properly tested! We should definitely organize those "testing sessions" as you suggested.

**Comment by mariehbourget** [comment]
> * I encountered a user scenario that has a high risk of resulting in data error. Currently, if a user loads an image in FSLeyes and then begins the segmentation, if a pixel size file doesn't exist the user will be prompted to enter the resolution. As we've suggested to some users, and I had to do myself, tweaking this parameter is sometimes necessary to adjust the patch size to improve the segmentations. **However**, if after the segmentation is finished the user clicks the save morphometrics button within the same session, **the user will not be prompted to enter the actual resolution, thus the values in the morphometrics file will be wrong (except for g-ratio)**. I caught it, but users likely wouldn't. My solution was to generate the segmentations for each images, close FSLeyes, and then load images one by one to manually fix them. I think we urgently need a solution to both of these problems mentioned above.

I’ve just realized that this will be an issue with the integration of the peripheral nerves segmentation model in FSLeyes.

We already know that the pixel size must be adjusted to improve the segmentation depending on image content. To compute the morphometrics accurately, the user will have to do as Mathieu suggested and close the session between segmentation and morphometrics.

Maybe we could address this in the settings menu (#458) with a pixel size field?
